src/main/java/org/graylog/integrations/notifications/types/microsoftteams/TeamsMessage.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2020 Graylog, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the Server Side Public License, version 1,
+ * as published by MongoDB, Inc.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.integrations.notifications.types.microsoftteams;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TeamsMessage {
```

Comment:
This looks like a simple POJO that maps the Java object to JSON. In most of the Graylog codebase, that is done with AutoValue classes (with annotated fields and create or builder methods) and Jackson and not with constructor-driven POJOs that have built-in serialization logic (see the `getJsonString()` method).

For consistency with the most of the pre-existing codebase, I suggest that we should use an AutoValue object instead, and post that directly with the OkHttp client and let the conversion to JSON happen implicitly, or manually convert it using Jackson. 

Here is an example:
https://github.com/Graylog2/graylog-plugin-enterprise/blob/72b5428c61cffe7c483e2363322222abc1701105/enterprise/src/main/java/org/graylog/plugins/license/LicenseReportPeriodical.java#L122-L143

Reply:
Now I am seeing that this was previously established in the Slack notification. https://github.com/Graylog2/graylog-plugin-integrations/blob/ac5718a038a3025474e8b7d18b2dfc609a91e7b0/src/main/java/org/graylog/integrations/notifications/types/SlackMessage.java#L33

I am fine with merging this PR as-is, but I think this should be refactored in a subsequent PR for consistency if possible. I think it would be best to avoid replicating this pattern, unless there is a reason for it I am not seeing here.

Reply:
@kingzacko1 and I discussed that this fits well in a separate PR after this is merged. It will be easier to review that way.

Reply:
Adding this here as because we're discussing a separate PR involving Slack: Webhook validation for Slack doesn't appear to working. I was able to (accidently) use a Teams webhook url in a Slack Graylog Notification, and receive a message in Microsoft Teams.